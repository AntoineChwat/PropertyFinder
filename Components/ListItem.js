'use strict';

const React = require('react');

const ReactNative = require('react-native');
const StyleSheet = ReactNative.StyleSheet;
const Image = ReactNative.Image;
const View = ReactNative.View;
const TouchableHighlight = ReactNative.TouchableHighlight;
const Text = ReactNative.Text;

const createReactClass = require('create-react-class');

const ListItem = createReactClass({
  _onPress: function() {
    this.props.onPressItem(this.props.index, this.props.item);
  },
  //Catch undefined fields
  render: function() {
    const self = this;
    return(
      <TouchableHighlight
        onPress={self._onPress}
        underlayColor='#dddddd'>
        <View>
          <View style={styles.rowContainer}>
            <Image style={styles.thumb} source={{ uri: self.props.item.img_url }} />
            <View style={styles.textContainer}>
              <Text style={styles.price}>{self.props.price}</Text>
              <Text style={styles.title}
                numberOfLines={1}>{self.props.item.title}</Text>
            </View>
          </View>
          <View style={styles.separator}/>
        </View>
      </TouchableHighlight>
    )
  }
})

const styles = StyleSheet.create({
  thumb: {
    width: 80,
    height: 80,
    marginRight: 10
  },
  textContainer: {
    flex: 1
  },
  separator: {
    height: 1,
    backgroundColor: '#dddddd'
  },
  price: {
    fontSize: 25,
    fontWeight: 'bold',
    color: '#48BBEC'
  },
  title: {
    fontSize: 20,
    color: '#656565'
  },
  rowContainer: {
    flexDirection: 'row',
    padding: 10
  },
});

module.exports = ListItem;
