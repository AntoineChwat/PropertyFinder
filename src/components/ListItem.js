'use strict';

const React = require('react');

const ReactNative = require('react-native');
const StyleSheet = ReactNative.StyleSheet;
const Image = ReactNative.Image;
const View = ReactNative.View;
const TouchableHighlight = ReactNative.TouchableHighlight;
const Text = ReactNative.Text;

const PropTypes = require('prop-types');

const createReactClass = require('create-react-class');

const ListItem = createReactClass({
  propTypes: {
    onPressItem: PropTypes.func.isRequired,
    index: PropTypes.number,
    item: PropTypes.object.isRequired
  },

  _onPress: function() {
    this.props.onPressItem(this.props.index, this.props.item);
  },
  render: function() {
    const self = this;
    var url = self.props.item.img_url;
    if (typeof url != 'string' || url =='') {
      url = 'https://www.pixedelic.com/themes/geode/demo/wp-content/uploads/sites/4/2014/04/placeholder4.png';
    }
    var price = self.props.item.price_formatted;
    if (typeof price != 'string' || price =='') {
      price = 'Unknown price';
    }
    var title = self.props.item.title;
    if (typeof title != 'string' || title =='') {
      title = 'Unknown title';
    }
    return (
      <TouchableHighlight
        onPress={self._onPress}
        underlayColor='#dddddd'>
        <View>
          <View style={styles.rowContainer}>
            <Image style={styles.thumb} source={{ uri: url }} />
            <View style={styles.textContainer}>
              <Text style={styles.price}>{price}</Text>
              <Text style={styles.title}
                numberOfLines={1}>{title}</Text>
            </View>
          </View>
          <View style={styles.separator}/>
        </View>
      </TouchableHighlight>
    );
  }
});

const styles = StyleSheet.create({
  thumb: {
    width: 80,
    height: 80,
    marginRight: 10
  },
  textContainer: {
    flex: 1
  },
  separator: {
    height: 1,
    backgroundColor: '#dddddd'
  },
  price: {
    fontSize: 25,
    fontWeight: 'bold',
    color: '#48BBEC'
  },
  title: {
    fontSize: 20,
    color: '#656565'
  },
  rowContainer: {
    flexDirection: 'row',
    padding: 10
  }
});

module.exports = ListItem;
