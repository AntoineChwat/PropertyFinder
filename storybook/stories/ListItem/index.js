const React = require('react');
const Component = React.Component;

const ReactNative = require('react-native');
const StyleSheet = ReactNative.StyleSheet;
const Image = ReactNative.Image;
const View = ReactNative.View;
const TouchableHighlight = ReactNative.TouchableHighlight;
const Text = ReactNative.Text;

const createReactClass = require('create-react-class');

const ListItem = require('../../../Components').ListItem;

const ListItemTest = createReactClass({
  render: function() {
    const item = {
      title: "Hello there",
      img_url: "https://lumiere-a.akamaihd.net/v1/images/Qui-Gon-Jinn_d89416e8.jpeg?region=0%2C1%2C1536%2C864&width=768"
    }
    return(
      <ListItem
        item = {item}
        price = {"General Kenobi"}
        onPressItem = {() => console.log("pressed")}
      />
    )
  }
})

const styles = StyleSheet.create({
  thumb: {
    width: 80,
    height: 80,
    marginRight: 10
  },
  textContainer: {
    flex: 1
  },
  price: {
    fontSize: 25,
    fontWeight: 'bold',
    color: '#48BBEC'
  },
  title: {
    fontSize: 20,
    color: '#656565'
  },
  rowContainer: {
    flexDirection: 'row',
    padding: 10
  },
});

module.exports = ListItemTest;
