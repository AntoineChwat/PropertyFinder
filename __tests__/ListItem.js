import 'react-native';
import React from 'react';
import ListItem from '../src/Components/ListItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import { configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import { shallow } from 'enzyme';

import sinon from 'sinon';

const item = {
	bathroom_number: '',
	bedroom_number: 1,
	car_spaces: 0,
	commission: 0,
	construction_year: 0,
	datasource_name: 'NetHousePrices',
	img_height: 300,
	img_url: 'https://imgs.nestimg.com/unspecified_london_108799203763910940.jpg',
	img_width: 400,
	keywords: '',
	latitude: 51.532856,
	lister_url: 'https://www.nestoria.co.uk/detail/0000000108799203763910940/title/5/1-1?serpUid=&pt=1&ot=1&l=london&did=90_high1&utm_source=api&utm_medium=external',
	listing_type: 'buy',
	location_accuracy: 9,
	longitude: -0.1351723,
	price: 545000,
	price_currency: '£',
	price_formatted: '£545,000',
	price_high: 545000,
	price_low: 545000,
	property_type: 'flat',
	size: 0,
	size_type: 'net',
	summary: '1 bedroom flat chain free long lease location immediately available...',
	thumb_height: 60,
	thumb_url: 'https://imgs.nestimg.com/medium/unspecified_london_108799203763910940.jpg',
	thumb_width: 80,
	title: 'Werrington Street, London, NW1',
	updated_in_days: 1,
	updated_in_days_formatted: 'new!'
};

describe('ListItem', () => {
	it('should render correctly', () => {
		const tree = renderer.create(
			<ListItem
				item={item}
				onPressItem={() => console.log('pressed')}
			/>
		);
		tree.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should be clickable', () => {
		const onPressSpy = sinon.spy();
		const wrapper = shallow(
			<ListItem
				item={item}
				onPressItem={onPressSpy}
			/>
		);
		wrapper.find('TouchableHighlight').simulate('press');
		expect(onPressSpy.calledOnce).toBe(true);
	});
	it('should handle an empty item', () => {
		const item = {};
		const wrapper = shallow(
			<ListItem
				item={item}
				onPressItem={() => {}}
			/>
		);
		const title = wrapper.find('Text').get(1).props.children;
		const price = wrapper.find('Text').get(0).props.children;
		const image = wrapper.find('Image').props().source.uri;
		expect(title).toMatch('Unknown title');
		expect(price).toMatch('Unknown price');
		expect(image).toMatch('https://www.pixedelic.com/themes/geode/demo/wp-content/uploads/sites/4/2014/04/placeholder4.png');
	});
});
