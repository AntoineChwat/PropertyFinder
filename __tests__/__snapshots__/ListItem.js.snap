// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={null}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://imgs.nestimg.com/unspecified_london_108799203763910940.jpg",
          }
        }
        style={
          Object {
            "height": 80,
            "marginRight": 10,
            "width": 80,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#48BBEC",
              "fontSize": 25,
              "fontWeight": "bold",
            }
          }
        >
          Â£545,000
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#656565",
              "fontSize": 20,
            }
          }
        >
          Werrington Street, London, NW1
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#dddddd",
          "height": 1,
        }
      }
    />
  </View>
</View>
`;
